middle_end/flambda2/simplify/simplify_let_expr.ml

```diff
@@ -328,6 +328,21 @@ let simplify_let0 ~simplify_expr ~simplify_function_body dacc let_expr
     let dacc =
       DA.add_to_lifted_constant_accumulator dacc prior_lifted_constants
     in
+    let dacc =
+      List.fold_left
+        (fun dacc (binding : Expr_builder.binding_to_place) ->
+          Bound_pattern.fold_all_bound_vars binding.let_bound ~init:dacc
+            ~f:(fun dacc bound_var ->
+              let v = Bound_var.var bound_var in
+              DA.map_denv dacc ~f:(fun denv ->
```

Comment:
Maybe do the `DA.map_denv` outside of the fold to avoid allocating too many dacc